packages/nextjs/src/utils/config.ts

```diff
@@ -1,94 +1,110 @@
-/* eslint-disable @typescript-eslint/no-non-null-assertion */
-/* eslint-disable @typescript-eslint/no-var-requires */
-/* eslint-disable @typescript-eslint/no-explicit-any */
 import { getSentryRelease } from '@sentry/node';
 import { logger } from '@sentry/utils';
 import defaultWebpackPlugin, { SentryCliPluginOptions } from '@sentry/webpack-plugin';
 import * as SentryWebpackPlugin from '@sentry/webpack-plugin';
-import * as fs from 'fs';
 import * as path from 'path';
 
-/**
- * Starting at `startPath`, move up one directory at a time, searching for `searchFile`.
- *
- * @param startPath The location from which to start the search.
- * @param searchFile The file to search for
- * @returns The absolute path of the file, if it's found, or undefined if it's not
- */
-function findUp(startPath: string, searchFile: string): string | undefined {
-  if (!fs.existsSync(startPath)) {
-    throw new Error(`The given \`startPath\` value (${startPath}) does not exist.`);
-  }
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type PlainObject<T = any> = { [key: string]: T };
+
+// Man are these types hard to name well. "Entry" = an item in some collection of items, but in our case, one of the
+// things we're worried about here is property (entry) in an object called... entry. So henceforth, the specific
+// proptery we're modifying is going to be known as an EntryProperty.
+
+// The function which is ultimately going to be exported from `next.config.js` under the name `webpack`
+type WebpackExport = (config: WebpackConfig, options: WebpackOptions) => WebpackConfig;
+
+// The two arguments passed to the exported `webpack` function, as well as the thing it returns
+// TODO use real webpack types
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+type WebpackConfig = { devtool: string; plugins: PlainObject[]; entry: EntryProperty; module: { rules: any[] } };
+type WebpackOptions = { dev: boolean; isServer: boolean };
```

Comment:
This one should be described in the `@types/webpack` also

Reply:
Same here.